docs-preview/src/extension.ts

```diff
@@ -35,12 +35,217 @@ export function activate(context: ExtensionContext) {
             const workingPath = filePath.replace(basename(filePath), "");
             return md.use(require("markdown-it-include"), { root: workingPath, includeRe: INCLUDE_RE })
                 .use(codeSnippets, { root: workingPath })
-                .use(xref);
-        }
+                .use(xref)
+                .use(custom_codeblock)
+                .use(column_end)
+                .use(container_plugin, "row", {
+                    marker: ":",
+                    validate(params) {
```

Comment:
this is really cool! Maybe we can clean this up a little? extension.ts is starting to look a little heavy...